Shared/utilities/GeoElementUtils.cpp

```diff
@@ -0,0 +1,135 @@
+
+#include "GeoElementUtils.h"
+
+// C++ API headers
+#include "GeoElement.h"
+#include "Feature.h"
+#include "Graphic.h"
+#include "EncFeature.h"
+#include "WmsFeature.h"
+#include "KmlPlacemark.h"
+
+using namespace Esri::ArcGISRuntime;
+
+namespace Dsa {
+
+/*!
+  \class Dsa::GeoElementSignaler
+  \brief A lightweight class that wraps a GeoElement and provides the geometryChanged signal.
+
+  This object does not take ownership of the GeoElement. You can choose to do that
+  by passing in the managed geoElement as the parent.
+
+  \sa Dsa::GeoElementUtils::toQObject
+ */
+
+/*!
+  \brief Constructor that takes a \a geoElement and an optional \a parent.
+ */
+GeoElementSignaler::GeoElementSignaler(GeoElement* geoElement, QObject* parent) :
+  QObject(parent),
+  m_geoElement(geoElement)
+{
+  if (dynamic_cast<Feature*>(m_geoElement))
+  {
+    connect(static_cast<Feature*>(m_geoElement), &Feature::geometryChanged,
+            this, &GeoElementSignaler::geometryChanged);
+  }
+  else if (dynamic_cast<Graphic*>(m_geoElement))
+  {
+    connect(static_cast<Graphic*>(m_geoElement), &Graphic::geometryChanged,
+            this, &GeoElementSignaler::geometryChanged);
+  }
+  else if (dynamic_cast<KmlPlacemark*>(m_geoElement))
+  {
+    connect(static_cast<KmlPlacemark*>(m_geoElement), &KmlPlacemark::geometryChanged,
+            this, &GeoElementSignaler::geometryChanged);
+  }
+  else if (dynamic_cast<EncFeature*>(m_geoElement))
+  {
+    connect(static_cast<EncFeature*>(m_geoElement), &EncFeature::geometryChanged,
+            this, &GeoElementSignaler::geometryChanged);
+  }
+  else if (dynamic_cast<WmsFeature*>(m_geoElement))
+  {
+    connect(static_cast<WmsFeature*>(m_geoElement), &WmsFeature::geometryChanged,
+            this, &GeoElementSignaler::geometryChanged);
+  }
+}
+
+/*!
+  \brief Destructor.
+ */
+GeoElementSignaler::~GeoElementSignaler() = default;
+
+/*!
+  \fn void Dsa::GeoElementUtils::setParent(const QList<Esri::ArcGISRuntime::GeoElement*> geoElements, QObject* parent)
+  \brief Allows the \a parent to be applied to all the \a geoElements.
+ */
+void GeoElementUtils::setParent(const QList<Esri::ArcGISRuntime::GeoElement*> geoElements, QObject* parent)
+{
+  if (geoElements.isEmpty())
+    return;
+
+  for (auto* geoElement : geoElements)
+  {
+    toQObject(geoElement)->setParent(parent);
+  }
+}
+
+/*!
+  \fn void Dsa::GeoElementUtils::setParent(Esri::ArcGISRuntime::GeoElement* geoElement, QObject* parent)
+  \brief Allows the \a parent to be applied to \a geoElement.
+ */
+void GeoElementUtils::setParent(Esri::ArcGISRuntime::GeoElement* geoElement, QObject* parent)
+{
+  if (geoElement)
+  {
+    toQObject(geoElement)->setParent(parent);
+  }
+}
+
+/*!
+  \fn QObject* Dsa::GeoElementUtils::toQObject(Esri::ArcGISRuntime::GeoElement* geoElement)
+  \brief Returns a QObject* from a GeoElement.
+ */
+QObject* GeoElementUtils::toQObject(Esri::ArcGISRuntime::GeoElement* geoElement)
+{
+  if (dynamic_cast<Feature*>(geoElement))
```

Comment:
I think you raised an issue for a "geo element type enum" - looks like it would make this a lot cleaner